packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1413,22 +1453,35 @@ export function attach(
     }
   }
 
+  // TRICKY
+  // This method recursively updates child order, but it does so iteratively.
+  // This is to avoid hitting the call stack limit for extremely deep or wide trees.
   function findReorderedChildrenRecursively(
     fiber: Fiber,
     nextChildren: Array<number>,
   ) {
-    if (!shouldFilterFiber(fiber)) {
-      nextChildren.push(getFiberID(getPrimaryFiber(fiber)));
-    } else {
-      let child = fiber.child;
-      while (child !== null) {
-        findReorderedChildrenRecursively(child, nextChildren);
-        child = child.sibling;
+    let fibers: Array<Fiber> = [fiber];
+
+    while (fibers.length > 0) {
+      const currentFiber = fibers.shift();
+
+      if (!shouldFilterFiber(currentFiber)) {
+        nextChildren.push(getFiberID(getPrimaryFiber(currentFiber)));
+      } else {
+        let child = currentFiber.child;
+        let childInsertionIndex = 0;
+        while (child !== null) {
+          fibers.splice(childInsertionIndex, 0, child);
+          childInsertionIndex++;
+          child = child.sibling;
+        }
       }
     }
   }
 
   // Returns whether closest unfiltered fiber parent needs to reset its child list.
+  //
+  // TODO Refactor this method to be iterative as well
   function updateFiberRecursively(
```

Comment:
This method needs to be migrated too, but it's not going to be as straight forward as the other two since it depends on its own recursive return value.